sdk/src/main/scala/com/horizen/account/state/ForgerStakeMsgProcessor.scala

```diff
@@ -183,65 +338,41 @@ object ForgerStakeMsgProcessor extends AbstractFakeSmartContractMsgProcessor {
           val stakeId : Array[Byte]          = cmdInput.stakeId
           val signature : SignatureSecp256k1 = cmdInput.signature
 
-          // get current list from db
-          val serializedStakeIdList : Array[Byte] = view.getAccountStorageBytes(fakeSmartContractAddress.address(), stakeIdsListKey).get
-
-          val stakeInfoList = serializedStakeIdList.length match {
-            case 0 =>
-              return new ExecutionFailed(RemoveStakeGasPaidValue, new Exception("No stakes in state-db"))
-
-            case _ =>  forgingInfoSerializer.parseBytesTry(serializedStakeIdList) match {
-              case Success(obj) => obj
-              case Failure(exception) =>
-                log.error("Error while parsing list of forging info.", exception)
-                return new InvalidMessage(new Exception(exception))
-            }
-          }
-
-          // remove the entry if any
-          var removedElement : AccountForgingStakeInfo = null
-          val newList = stakeInfoList.asScala.filterNot(
-            x => {
-              if (BytesUtils.toHexString(x.stakeId) == BytesUtils.toHexString(stakeId))
-              {
-                removedElement = x
-                true
-              } else {
-                false
-              }
-            })
-
-          if (removedElement == null) {
-            return new ExecutionFailed(RemoveStakeGasPaidValue, new Exception("No such stake id in state-db"))
+          // get the forger stake data to remove
+          val stakeData = findStakeData(view, stakeId) match {
```

Comment:
In this case it is a little different, because the correct path is the removal of the stake, and in that case we have to retrieve the data.
Only if a not-existing stakeId is passed we would have the optimization, but would call both api in the correct path

Reply:
ok, got it. Fine to me